### ISSUE TITLE
Report from Instructor - Student Cannot Achieve 100% Progress

### ISSUE THREAD

**Issue by kwahlenmayer** [Simon-Initiative/oli-torus#4212]
Report from Instructor:

On 2023-09-14 15:35:47.634079+00:00 UTC UTC, Avital Pelakh < avp36@pitt.edu > wrote:

Hi! I am a student of Ken Koedinger's using OLI to create a course. I just piloted my course by logging in as a guest and even though I answered all of the questions, my dashboard is telling me I'm only 80% complete.

Here is the link for the course section: https://proton.oli.cmu.edu/sections/join/33rk9

----------------------------------------------
Timestamp: 2023-09-14 15:35:47.634079+00:00 UTC UTC
Ip Address: 0.0.0.0.0.65535.32512.1
Location: https://proton.oli.cmu.edu/sections/reacting_to_errors__version_a/overview

WEB BROWSER
User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36
Accept: */*
Language: en-US,en;q=0.9

CAPABILITIES
Cookies Enabled: true

USER ACCOUNT
Name:
Email:
Created: September 14, 2023 8:58 AM

**Comment by kwahlenmayer** [comment]
Testing Report
 
I have 100% score on Scenario 1 and 2 (since I manually scored all of my student Wahlenmayer's instructor-scored items)
But Scenario 1 and 2 show as 0% on student completion
So my student  only shows 80% progress
How can my Scenario 1 and 2 go to 100% progress so my student shows as 100% progress?
None of the other students have 100% progress
https://proton.oli.cmu.edu/sections/reacting_to_errors__version_a/student_dashboard/5331/content
I am asking on behalf of an instructor of this course (original report above)
I'm using my own test student account among other students and guests





C
1 external person is from CMU

**Comment by darrensiegel** [comment]
This is a bug in graded page finalization:

```
def finalize(%FinalizationContext{
        resource_attempt: %ResourceAttempt{lifecycle_state: :active} = resource_attempt,
        section_slug: section_slug,
        datashop_session_id: datashop_session_id,
        effective_settings: effective_settings
      }) do
    # Collect all of the part attempt guids for all of the activities that get finalized
    with {:ok, part_attempt_guids} <-
           finalize_activity_and_part_attempts(resource_attempt, datashop_session_id),
         {:ok, resource_attempt} <- roll_up_activities_to_resource_attempt(resource_attempt, effective_settings),
         {:ok, resource_attempt} <- cancel_pending_auto_submit(resource_attempt) do
      case resource_attempt do
        %ResourceAttempt{lifecycle_state: :evaluated} ->
          case roll_up_resource_attempts_to_access(
                 effective_settings,
                 section_slug,
                 resource_attempt.resource_access_id,
                 resource_attempt.was_late
               ) do
            {:ok, resource_access} ->
              {:ok, resource_access} = Oli.Delivery.Metrics.mark_progress_completed(resource_access)

              {:ok,
               %FinalizationSummary{
                 graded: true,
                 lifecycle_state: :evaluated,
                 resource_access: resource_access,
                 part_attempt_guids: part_attempt_guids,
                 effective_settings: effective_settings
               }}

            error ->
              error
          end

        %ResourceAttempt{lifecycle_state: :submitted, resource_access_id: resource_access_id} ->
          {:ok,
           %FinalizationSummary{
             graded: true,
             lifecycle_state: :submitted,
             resource_access: Oli.Repo.get(ResourceAccess, resource_access_id),
             part_attempt_guids: part_attempt_guids,
             effective_settings: effective_settings
           }}
      end
    else
      error -> error
    end
  end
```

The call to `Oli.Delivery.Metrics.mark_progress_completed` only takes place when a page attempt is fully evaluated.  When rolled up to `:submitted` it is left untouched.  We don't update at the point of manual grading either, but this is the place that we should update it.  Progress credit should be given at point of time of answering question, not when the instructor eventually scores it.

**Comment by bitbldr** [comment]
Duplicate of a previously resolved issue.

**Comment by kwahlenmayer** [comment]
This image shows my previous enrollment (oli-help@cmu.edu) with **0% progress despite having 100% score** on Scenario 1 and 2 which are manually graded.
![image](https://github.com/Simon-Initiative/oli-torus/assets/24357828/74432ff0-66b6-49a6-827e-e12b92bc6e33)

**Comment by kwahlenmayer** [comment]
Testing this today with a different account, Wahlenmayer, Kim "Student 1", I find the same unexplained result:
100% score on scenario 1 and 2 (these have been manually scored by the instructor)
yet
0% progress on scenario 1 and 2

To replicate this finding:
enroll as a student
complete scenarios 1 and 2
at this point, these two scenarios should (arguably) be 100% complete but still waiting for instructor scoring
as instructor or admin, score the 4 items from scenarios 1 and 2
student scores on scenarios 1 and 2 are now complete (as expected)
yet - and here is the issue - the student progress is 0%

![image](https://github.com/Simon-Initiative/oli-torus/assets/24357828/a7c2d8ae-49b8-48f6-985f-0bc27a6b7993)

**Comment by kwahlenmayer** [comment]
Here is another image showing 0% progress on scenario 1 and 2:

![image](https://github.com/Simon-Initiative/oli-torus/assets/24357828/8c445462-f7ab-433d-8034-42a6edce9f85)

**Comment by kwahlenmayer** [comment]
Yes, here is another image that shows scores on scenarios 1 and 2:

![image](https://github.com/Simon-Initiative/oli-torus/assets/24357828/8e06e2f1-602b-4f1a-9320-dbd224a6346c)

**Comment by darrensiegel** [comment]
Closing, this is fixed but not yet in production. It is fixed in v25
